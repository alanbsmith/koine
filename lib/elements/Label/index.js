import styled from 'styled-components';
import { rem } from 'polished';
import { applyStyleModifiers } from 'styled-components-modifiers';

import { fontStyles, fontWeights } from '../../modifiers';

const modifierConfig = {
  ...fontStyles,
  ...fontWeights,
};


const Label = styled.label`
  color: ${props => props.theme.colors.base.chrome600};
  font-size: ${rem(14)};
  font-weight: 600;
  text-transform: capitalize;
  ${applyStyleModifiers(modifierConfig)}
`;

export default Label;
