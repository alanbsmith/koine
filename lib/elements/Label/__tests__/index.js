import React from 'react';
import Label from '../index';
import 'jest-styled-components';

import { renderWithTheme } from '__tests__/helpers';


describe('Label Element', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <Label>Label</Label>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the italicize modifier', () => {
    const tree = renderWithTheme(
      <Label modifiers={['italicize']}>Label</Label>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the fontWeightBold modifier', () => {
    const tree = renderWithTheme(
      <Label modifiers={['fontWeightBold']}>Label</Label>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
