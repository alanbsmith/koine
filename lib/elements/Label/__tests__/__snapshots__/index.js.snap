// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Element adds the fontWeightBold modifier 1`] = `
.c0 {
  color: #646464;
  font-size: 0.875rem;
  font-weight: 600;
  text-transform: capitalize;
}

<label
  className="c0"
>
  Label
</label>
`;

exports[`Label Element adds the italicize modifier 1`] = `
.c0 {
  color: #646464;
  font-size: 0.875rem;
  font-weight: 600;
  text-transform: capitalize;
  font-style: italic;
}

<label
  className="c0"
>
  Label
</label>
`;

exports[`Label Element renders correctly 1`] = `
.c0 {
  color: #646464;
  font-size: 0.875rem;
  font-weight: 600;
  text-transform: capitalize;
}

<label
  className="c0"
>
  Label
</label>
`;
